torch/nn/modules/rnn.py

```diff
@@ -18,6 +18,13 @@
 }
 
 
+def apply_permutation(tensor, permutation, contiguous_output, dim=1):
+    result = tensor.index_select(dim, permutation)
+    if contiguous_output:
```

Comment:
Question: don't we always return contiguous tensors from index_select?

Reply:
oh, you have a good point, I forgot about that